### ISSUE TITLE
Traefik redirecting to wrong host when empty host

### ISSUE THREAD

**Issue by kmeekva** [containous/traefik#6336]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
Bing bot hit URL that requires client certificate with no SNI hostname.

### What did you expect to see?
Traefik should NOT pass request to remote service container.

### What did you see instead?
Logs in our remote container show request was passed from traefik for this request.

`

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.1.2
Codename:     cantal
Go version:   go1.13.5
Built:        2020-01-07T18:23:57Z
OS/Arch:      linux/amd64

```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

<details>
<summary>cat traefik.yml</summary>

```yml
global:
  checkNewVersion: true
  sendAnonymousUsage: true
entryPoints:
  http:
    address: :80
  https:
    address: :443
  traefik:
    address: :8080
providers:
  providersThrottleDuration: 2s
  docker:
    watch: true
    endpoint: unix:///var/run/docker.sock
    exposedByDefault: false
    swarmMode: false
  file:
    directory: /etc/traefik/config
    watch: true

api:
  insecure: true
  dashboard: true
log:
  level: DEBUG
```

</details>

<details>
<summary>cat middlewares.yml</summary>

```yml
http:
  middlewares:
    https-redirect:
      redirectScheme:
        scheme: https

    security-headers:
      headers:
        sslRedirect: true
        accessControlAllowMethods:
          - GET
          - OPTIONS
          - PUT
        stsSeconds: 315360000
        stsIncludeSubdomains: true
        stsPreload: true
        customFrameOptionsValue: SAMEORIGIN
        referrerPolicy: same-origin
        featurePolicy: vibrate 'self'
        accessControlAllowOrigin: origin-list-or-null
        accessControlMaxAge: 100
        BrowserXssFilter: true
        contentTypeNosniff: true
        forceSTSHeader: true
        frameDeny: true
```

</details>

<details>
<summary>cat tls.yml</summary>

```yml
tls:
  options:
    tcgdefault:
      minVersion: VersionTLS12
      maxVersion: VersionTLS12
      clientAuth:
        caFiles:
          - /etc/traefik/config/ca-bundle.crt
        clientAuthType: RequireAndVerifyClientCert
    fastdefault:
      minVersion: VersionTLS12
      maxVersion: VersionTLS12
      clientAuth:
        caFiles:
          - /etc/traefik/config/fast-ca-bundle.crt
        clientAuthType: RequireAndVerifyClientCert
  stores:
    default:
      defaultCertificate:
        certFile: /etc/traefik/config/columbiagroup.com.crt
        keyFile: /etc/traefik/config/columbiagroup.com.key
```

</details>

<details>
<summary>Docker labels:</summary>

```yml
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=traefik"
      - "traefik.http.routers.router_${INSTANCE}.rule=Host(`${HOST}`) && PathPrefix(`/${INSTANCE}`)"
      - "traefik.http.middlewares.${INSTANCE}-stripprefix.stripprefix.prefixes=/${INSTANCE}"
      - "traefik.http.routers.router_${INSTANCE}.entrypoints=https"
      - "traefik.http.routers.router_${INSTANCE}.tls=true"
      - "traefik.http.routers.router_${INSTANCE}.tls.options=tcgdefault@file"
      - "traefik.http.routers.router_${INSTANCE}.middlewares=${INSTANCE}-stripprefix,${INSTANCE}-passtlsclientcert"

      - "traefik.http.routers.router_${INSTANCE}_dns.rule=Host(`${INSTANCE}.columbiagroup.com`)"
      - "traefik.http.routers.router_${INSTANCE}_dns.entrypoints=https"
      - "traefik.http.routers.router_${INSTANCE}_dns.tls=true"
      - "traefik.http.routers.router_${INSTANCE}_dns.tls.options=tcgdefault@file"
      - "traefik.http.routers.router_${INSTANCE}_dns.middlewares=${INSTANCE}-passtlsclientcert"

      - "traefik.http.routers.redirect_router_${INSTANCE}.rule=Host(`${HOST}`) && PathPrefix(`/${INSTANCE}`)"
      - "traefik.http.routers.redirect_router_${INSTANCE}.entrypoints=http"
      - "traefik.http.routers.redirect_router_${INSTANCE}.middlewares=https-redirect@file"

      - "traefik.http.routers.redirect_router_${INSTANCE}_dns.rule=Host(`${INSTANCE}.columbiagroup.com`)"
      - "traefik.http.routers.redirect_router_${INSTANCE}_dns.entrypoints=http"
      - "traefik.http.routers.redirect_router_${INSTANCE}_dns.middlewares=https-redirect@file"

      - "traefik.http.middlewares.${INSTANCE}-passtlsclientcert.passtlsclientcert.pem=true"
      - "traefik.http.middlewares.${INSTANCE}-passtlsclientcert.passtlsclientcert.info.subject.commonName=true"
      - "traefik.http.middlewares.${INSTANCE}-passtlsclientcert.passtlsclientcert.info=true"
```

</details>

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
time="2020-02-13T19:38:30Z" level=debug msg="Serving default certificate for request: \"\""
time="2020-02-13T19:38:31Z" level=warning msg="Tried to extract a certificate on a request without mutual TLS" middlewareName=tasker-passtlsclientcert@docker middlewareType=PassClientTLSCert
time="2020-02-13T19:38:31Z" level=warning msg="Tried to extract a certificate on a request without mutual TLS" middlewareName=tasker-passtlsclientcert@docker middlewareType=PassClientTLSCert
time="2020-02-13T19:38:31Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/robots.txt\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"*/*\"],\"Accept-Encoding\":[\"gzip, deflate\"],\"Cache-Control\":[\"no-cache\"],\"Connection\":[\"Keep-Alive\"],\"Pragma\":[\"no-cache\"],\"User-Agent\":[\"Mozilla/5.0 (compatible; bingbot/2.0; +http://www.bing.com/bingbot.htm)\"],\"X-Forwarded-Host\":[\"XXX.columbiagroup.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"4276668a6949\"],\"X-Real-Ip\":[\"157.55.39.43\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"XXX.columbiagroup.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"157.55.39.43:11042\",\"RequestURI\":\"/robots.txt\",\"TLS\":null}"
```

**Comment by mpl** [comment]
Sorry, but we do not understand what you are trying to do. Can you restate better your use-case and how you are trying to address it?

**Comment by kmeekva** [comment]
Our site requires a client certificate to access it -- tls options clientAuthType is set to RequireAndVerifyClientCert.

But I noticed in the logs of our website running in a container behind traefik, that there were 404 errors for request from BING bot trying to request the /robots.txt file.

BING does not have a client certificate issued by our site.  So they should never be able to reach our back end service.   

I'm trying to understand what would allow traefik to pass something without a tls cert to this backend.  

The log entries above seem to indicate a request came in with NO hostname and somehow matched my host rule and was sent without TLS verification.

The middle ware is complaining about "Tried to extract a certificate on a request without mutual TLS"

But if there is no mutual TLS why would it send the request to the backend service???

**Comment by rtribotte** [comment]
Hi,

From what I can reproduce, the log output that mention certificate extraction without mutual TLS, occurs when the TLS conf is missing/wrong.

To go further, can you paste those kinds of log outputs:

`level=debug msg="Adding route for your.domain with TLS options tcgdefault@file" entryPointName=https`

`level=debug msg="Configuration received from provider docker:...`

Thanks

**Comment by kmeekva** [comment]
The TLS config this site is using "tcgdefault@file" is defined in our tls.yml and is working properly from what I can tell.   The site does prompt for client certificate and passes the certificate to the backend service which we use for authentication.   If a client certificate is not presented the connection is refused as we expect.

I think we found the problem.

We have 2 routers for each application.  One using a application specific host name and one using the underlying host name with a strip prefix.

I restarted Traefik and noticed this in the logs:

> time="2020-02-24T15:01:04Z" level=warning msg="Found different TLS options for routers on the same host dockerp1.columbiagroup.com, so using the default TLS options instead for these routers: []string{\"router_fast-qa@docker\", \"router_blast@docker\"}" entryPointName=https

When 2 different apps that tried to apply different TLS options ( to use a different CA bundle ) they were trying to assign these to the same host using strip prefix.  I see this is not allowed.  One Hostname must have one tls config.  So it was falling back to use the default TLS options instead.

We did not have a default defined in our tls.yml.  So the default is to not require client certificate.

I was able to reproduce the problem by accessing our site using the host with the path prefix and it did not prompt for the Client certificate.

Not sure how the BING spider found our application with the path prefix...  But that is another issue.

I modified the configuration to have a default TLS which requires client cert and it appears to use that and should resolve my issue.  

I'll fix the conflicts so that all path prefix routers use same defined TLS config, but good to know that a strong default is a good fall back.

Thanks for your help figuring this out.  

We Love Traefik!
